woocommerce/woocommerce-android
# [Blaze] Payment summary screen

Part of: #10721


### Description
This PR adds the following changes:
1. Updates FluxC to include the changes of https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2953
2. Adds UI models and Repository functions to fetch payment methods data.
3. Adds the payment summary screen.

The big number of lines impacted is because of new icons.

### Testing instructions
1. Sign in using WordPress.com.
2. Open the Blaze campaign creation flow.
3. When the AD details are loaded, tap on "Confirm Details".
5. Notice the payment summary screen.
6. Confirm the screen's behavior (the click on the payment method is not implemented yet).

### Images/gif
| Has saved payment methods | No saved payment methods |
| --- | --- |
| ![Screenshot_20240209_120925](https://github.com/woocommerce/woocommerce-android/assets/1657201/abde5428-f6e4-41ac-9526-fab1b8f015ce) | ![Screenshot_20240209_121046](https://github.com/woocommerce/woocommerce-android/assets/1657201/b94e2489-9ab2-4643-9773-712dc1cfd4a3) |
